### PR TITLE
RDCC-872 addressing review comments, removed mocks and added verify(s)

### PR DESCRIPTION
**Pull request by AdilOoze**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-872

### Change description ###

Addressing review comments

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[✔︎] No
```

### CODE REVIEW

File: config/owasp/suppressions.xml [hunk 1]
```diff
@@ -2234,6 +2234,76 @@
     </suppress>
     <suppress>
         <notes><![CDATA[
+   file name: htmlunit-driver-2.36.0.jar
```
Comment (shreedhar-hmcts) [inline config/owasp/suppressions.xml]: Remove these CVEs, check Abhijit's email for this.

Comment (AdilOoze) [reply config/owasp/suppressions.xml]: They have been removed in this commit: https://github.com/hmcts/rd-user-profile-api/pull/249/commits/4774a1cdadb7db0645df942a7dcd38a781f9f52a

File: config/owasp/suppressions.xml [hunk 2]
```diff
@@ -2234,6 +2234,41 @@
...
+    </suppress>
+    <suppress>
+        <notes><![CDATA[
+   file name: htmlunit-cssparser-1.5.0.jar
+   ]]></notes>
+        <sha1>7e40e04f32622e63a30e92ac158f799f67d8ddf0</sha1>
+        <cpe>cpe:/a:htmlunit_project:htmlunit</cpe>
+    </suppress>
+    <suppress>
+        <notes><![CDATA[
```
Comment (shashank-rastogi) [inline config/owasp/suppressions.xml]: even for html-unit.jar - u dont need so many suppressions

File: src/integrationTest/java/uk/gov/hmcts/reform/userprofileapi/integration/RetrieveUserProfileInternalServerErrorIntTest.java [hunk 3]
```diff
@@ -60,15 +60,15 @@ public void setUp() {
     public void should_return_500_and_not_create_user_profile_when_idam_service_throws_exception() throws Exception {
 
         when(idamService.registerUser(any(IdamRegisterUserRequest.class)))
-            .thenThrow(new RuntimeException("This is a test exception"));
+                .thenThrow(new RuntimeException("This is a test exception"));
```
Comment (shashank-rastogi) [inline src/integrationTest/java/uk/gov/hmcts/reform/userprofileapi/integration/RetrieveUserProfileInternalServerErrorIntTest.java]: why the message is "This is a test exception"

File: src/integrationTest/java/uk/gov/hmcts/reform/userprofileapi/integration/RetrieveUserProfileInternalServerErrorIntTest.java [hunk 4]
```diff
@@ -81,64 +81,36 @@ public void should_return_500_when_repository_throws_an_unknown_exception() thro
         IdamRegisterUserRequest request = Mockito.mock(IdamRegisterUserRequest.class);
         IdamRegistrationInfo idamRegistrationInfo = new IdamRegistrationInfo(CREATED);
         when(idamService.registerUser(request))
-            .thenReturn(idamRegistrationInfo);
+                .thenReturn(idamRegistrationInfo);
         when(userProfileRepository.findByIdamId(any(String.class)))
-            .thenThrow(new RuntimeException("This is a test exception"));
+                .thenThrow(new RuntimeException("This is a test exception"));
```
Comment (shashank-rastogi) [inline src/integrationTest/java/uk/gov/hmcts/reform/userprofileapi/integration/RetrieveUserProfileInternalServerErrorIntTest.java]: why the message is "This is a test exception"

File: src/integrationTest/java/uk/gov/hmcts/reform/userprofileapi/integration/RetrieveUserProfileInternalServerErrorIntTest.java (hunk 3 above)
Comment (AdilOoze) [reply src/integrationTest/java/uk/gov/hmcts/reform/userprofileapi/integration/RetrieveUserProfileInternalServerErrorIntTest.java]: I have updated the message to reflect the exception being thrown, in this case Runtime Exception, in this commit: https://github.com/hmcts/rd-user-profile-api/pull/249/commits/144afe4b496160730449f10772744c491b34d1cb

File: src/integrationTest/java/uk/gov/hmcts/reform/userprofileapi/integration/RetrieveUserProfileInternalServerErrorIntTest.java (hunk 4 above)
Comment (AdilOoze) [reply src/integrationTest/java/uk/gov/hmcts/reform/userprofileapi/integration/RetrieveUserProfileInternalServerErrorIntTest.java]: I have updated the message to reflect the exception being thrown, in this case Runtime Exception, in this commit: https://github.com/hmcts/rd-user-profile-api/pull/249/commits/144afe4b496160730449f10772744c491b34d1cb

File: config/owasp/suppressions.xml (hunk 2 above)
Comment (AdilOoze) [reply config/owasp/suppressions.xml]: I have removed all suppressions and placed only one for html-unit.jar, in this commit: https://github.com/hmcts/rd-user-profile-api/pull/249/commits/144afe4b496160730449f10772744c491b34d1cb

File: config/owasp/suppressions.xml [hunk 5]
```diff
@@ -2234,6 +2234,13 @@
     </suppress>
     <suppress>
         <notes><![CDATA[
+    file name: htmlunit-cssparser-1.5.0.jar
+    ]]></notes>
+        <packageUrl regex="true">^pkg:maven/net\.sourceforge\.htmlunit/htmlunit\-cssparser@.*$</packageUrl>
+        <cve>CVE-2020-5529</cve>
+    </suppress>
+    <suppress>
+        <notes><![CDATA[
```
Comment (AdilOoze) [inline config/owasp/suppressions.xml]: suppressing css parser as it is the latest version, so upgrade is not an option.

File: src/functionalTest/java/uk/gov/hmcts/reform/userprofileapi/AbstractFunctional.java [hunk 6]
```diff
@@ -1,8 +1,8 @@
 package uk.gov.hmcts.reform.userprofileapi;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static uk.gov.hmcts.reform.userprofileapi.helper.CreateUserProfileDataTestBuilder.buildCreateUserProfileData;
-import static uk.gov.hmcts.reform.userprofileapi.helper.CreateUserProfileDataTestBuilder.buildUpdateUserProfileData;
+import static uk.gov.hmcts.reform.userprofileapi.helper.CreateUserProfileTestDataBuilder.buildCreateUserProfileData;
+import static uk.gov.hmcts.reform.userprofileapi.helper.CreateUserProfileTestDataBuilder.buildUpdateUserProfileData;
```
Comment (AdilOoze) [inline src/functionalTest/java/uk/gov/hmcts/reform/userprofileapi/AbstractFunctional.java]: Name changes as requested by @mallikarjunputtana

**Pull request merged**